File: cash.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>

int main(void)
{
    float money = get_float("Change owed: ");
   if (money < 0)
    {
         money = get_float("Change owed: ");
    }
    //making sure that the input is not negative
    int coins = round(money * 100);
    //converting the dollars to coins
    int i = 0;
    // implementing a counter for number of coins to be returned
    while (coins >= 25)
    {
        coins = coins - 25;
        i++;
    }
    //seeing how many quarters can be given back
    while (coins >= 10)
    {
        coins = coins - 10;
        i++;
    }
    //seeing how many dimes can be given back
    while (coins >= 5)
    {
        coins = coins - 5;
        i++;
    }
    //seeing how many nickels can be given back
    i = i + coins;
    //adding remaining pennies to coins to be returned
    printf("%i\n", i);
}

File: cash.py
from cs50 import get_float

while True:
    n = get_float("Change owed: ")
    if n > 0:
        break
# converting the money amount into coins
n *= 100
coins = 0

# flowing through everything to count the total number of coins
while n >= 25:
    n -= 25
    coins += 1
while n >= 10:
    n -= 10
    coins += 1
while n >= 5:
    n -= 5
    coins += 1
# printing the final number of coins owed
print(f"{int(coins + n)}")
